contracts/src/Teleporter/ReceiptQueue.sol

```diff
@@ -37,25 +39,26 @@ contract ReceiptQueue {
     error Unauthorized();
     error EmptyQueue();
 
-    constructor() {
-        owner = msg.sender;
-    }
-
     /**
      * @dev Adds a receipt to the queue.
      *
      * Requirements:
      *
      * - `msg.sender` must be the owner.
      */
-    function enqueue(TeleporterMessageReceipt calldata receipt) external {
-        if (msg.sender != owner) {
-            revert Unauthorized();
+    function enqueue(
+        TeleporterMessageReceiptQueue storage queue,
```

Comment:
Is `storage` specified here in order to mutate the queue passed by the caller?

Reply:
it has to be storage because `TeleporterMessageReceiptQueue` contains a mapping